src/Voron/Impl/Scratch/ScratchBufferPool.cs

```diff
@@ -81,6 +81,12 @@ public ScratchBufferPool(StorageEnvironment env)
 
                     _recycleArea.RemoveFirst();
                 }
+
+                if (_current.File.IsDisposed == false)
+                {
+                    _current.File.Dispose();
+                    _scratchSpaceMonitor.Decrease(_current.File.NumberOfAllocatedPages * Constants.Storage.PageSize);
+                }
```

Comment:
It's safe to add this but I don't think it's the root cause and might not guarantee `_current` will be _always_ disposed.

The current implementation assumes that `_current` file should be on `_scratchBuffers` dict - see `AddScratchBufferFile()` method.

I wonder if we might have a race condition between iterating over `_scratchBuffers` and `NextFile()` call - `NextFile()` called at the time of `ScratchBufferPool` being disposed? 

The only scenario I could think about is that `NextFile()` is called from `Free()` which is called in the background by the flusher - so maybe concurrently with environment dispose?

Reply:
Yes, my fix is wrong, I forgot what I found before and even wrote about in the issue...
The problem is that scratch file dispose of isn't cleaning the dictionaries and the _current reference is keeping those dictionaries alive